api/src/main/java/org/datacleaner/api/RestrictedFunctionalityCallToAction.java

```diff
@@ -0,0 +1,102 @@
+/**
+ * DataCleaner (community edition)
+ * Copyright (C) 2014 Neopost - Customer Information Management
+ *
+ * This copyrighted material is made available to anyone wishing to use, modify,
+ * copy, or redistribute it subject to the terms and conditions of the GNU
+ * Lesser General Public License, as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+ * or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU Lesser General Public License
+ * for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this distribution; if not, write to:
+ * Free Software Foundation, Inc.
+ * 51 Franklin Street, Fifth Floor
+ * Boston, MA  02110-1301  USA
+ */
+package org.datacleaner.api;
+
+import java.io.Serializable;
+
+/**
+ * Represents a "Call to Action" for users when restricted functionality is
+ * blocked, typically with the use of {@link RestrictedFunctionalityException}
+ * or {@link RestrictedFunctionalityMessage}.
+ */
+public class RestrictedFunctionalityCallToAction implements Serializable {
+
+    private static final long serialVersionUID = 1L;
+
+    private final String _name;
```

Comment:
Subclasses could also work too, instead of enums. Under normal situations, the only ones I can think of we would use right now are "upgrade to edition X" and "buy credits". Maybe in the future we would have something for user to buy premium versions of certain transformers, eg app store like. The most extreme scenario would probably be a third party reinventing the entire credit system (although they would probably just create a fork in that case).

Maybe turn it in an interface then and have implementing classes for the two scenarios we need right now?